_knitr/A Shiny app for Hong Kong District Councillors_20200801.md

```diff
@@ -8,43 +8,160 @@ output:                    # DO NOT CHANGE
+
+- [shinydashboard](https://github.com/rstudio/shinydashboard): For mobile-friendly dashboard layout.
+    + We understood that our users, primarily HK citizens, frequently use mobiles. Thus, to ensure this app was useful to them, we centred our design on how the app looked on their mobile browsers.
+    
+<img src="https://raw.githubusercontent.com/martinctc/blog/master/images/googlesheets4.png" width=200>
+   
+- [googlesheets4](https://github.com/tidyverse/googlesheets4): For seamless access to Google Sheets.
+    + We understood that our users are not all technical so we stored the core data in a format and platform familiar and accessible to most people, Google Sheets.
+    + At a later stage of the app development, we migrated to storing the data in an R package we wrote, called [**hkdatasets**](https://github.com/hong-Kong-Districts-Info/hkdatasets) as we sought to keep the data in one place. However, the Google Sheets implementation worked very well, and the app could be deployed with no impact on performance or user experience.
+    
+- [sf](https://github.com/r-spatial/sf) and [leaflet](https://github.com/rstudio/leaflet): For importing geographic data and creating interactive maps.
+    + We understood that our users may want to explore other parts of Hong Kong but may not know the names of each constituency. Thus, we provided a map functionality to improve the ease they can learn more about different parts of Hong Kong.
+
+- [rintrojs](https://github.com/carlganz/rintrojs): For interactive tutorials.
+    + We understood that our users are not necessarily keen to read pages of instructions on how to use the app, especially if they are on mobile. Thus, we implemented a dynamic feature that walks them through visually each component of the app.
+
+## 🗄️ How was the data collected?
+
+Since there was no existing single data source on the DCs, we had to put this togther ourselves. All the data on each District Councillor, their constituency, the party they belong to, and their Facebook page was all collected manually through a combination of Wikipedia and Facebook. The data was initially housed on Google Sheets, for multiple reasons: 
+
+1. Using Google Sheets made it easy for multiple people to collaborate on data entry.
+2. Keeping the data outside of the repo has the advantage of keeping the memory size minimal, in line with best practices.
+3. By storing the data in Google Sheets, non-technical users would also be able to access the data too.
+
+Most of all, it was plain easy to access the Google Sheets data with the {googlesheets4} package! A key function is `googlesheets4::gs4_auth()`, which directs the developer to a web browser, asked to sign in to their Google account, and to grant googlesheets4 permission to operate on their behalf with Google Sheets. This Google Sheet is available [here](https://docs.google.com/spreadsheets/d/1007RLMHSukSJ5OfCcDJdnJW5QMZyS2P-81fe7utCZwk/).
```

Comment:
Worth noting that for the app to work, we actually had to use `googlesheets4::gs4_deauth()` to access the public Google Sheet in a de-authorised state. This is because the Shiny app itself does not have any particular Google credentials stored alongside it (and it probably shouldn't for security reasons). 

It was possible because the GSheet was available to anyone with the link. If it wasn't then we would need authorisation.

At the same time, you do need to authorise to edit the data (since only certain people can edit it). Thus, I think it's worth mentioning the subtle distinction here.